node/src/actors/chain_manager/mining.rs

```diff
@@ -111,6 +112,13 @@ impl ChainManager {
             _ => return,
         };
 
+        // Check eligibility
+        // S(H(beacon))
+        let mut vrf_input = match self.handle(GetHighestVrfOutput, ctx) {
+            Ok(b) => b,
+            _ => return,
+        };
```

Comment:
IMO `self.handle()` is code smell for all cases where the handler discards the context.

I'd move the logic from the handler into a `function get_highes_vrf_output` in `impl ChainManager` and then invoke that function from the handler and here.

This gives us better testability, as it allows unit testing the logic against the minimal set of data it is set to operate on (`struct ChainManager`), without intrinsics of the actor system getting in the middle (e.g. `ctx`).  

P.S. I know `AddTransaction` is also being used through `self.handle()`. I think we wasn't aware of these concerns at the time we wrote that. Feel free to leave it like it is or make the same change there too.

Reply:
What do you think about this?

Reply:
I am on it, trying to understand it first hehe.

Reply:
If we are using form ChainManager (mining is inside of ChainManager). Why dont use the field 'chain_info.highest_block_checkpoint' instead of a handler?

Reply:
Oops it is true that `GetHighestVrfOutput` is not being used from any other actor.

I'd drop the handler altogether but create the getter as proposed before just in case in the future we need to add the handler back or use that logic somewhere else.

Reply:
Should I change the way we get the beacon as well then?

Reply:
I mean from the state of the chain manager actor as  'chain_info.highest_vrf_checkpoint' and  'chain_info.highest_block_checkpoint'

Reply:
> Should I change the way we get the beacon as well then?

Yes please